rfcs/0005-pablo-distribution-assets/0005-pablo-distribution.adoc

```diff
@@ -0,0 +1,632 @@
+----
+skinparam usecase {
+BackgroundColor<< Staking >> YellowGreen
+BorderColor<< Staking >> YellowGreen
+
+ArrowColor Olive
+}
+left to right direction
+actor LP as lp
+actor Governance as g
+actor PicaStaker as s
+actor PabloStaker as p
+
+package Picasso {
+    package Pablo {
+        usecase "Stake PBLO" as UC2 <<Staking>>
+        usecase "Earn PBLO Reward" as UC4 <<Staking>>
+        usecase "Stake LP Token" as UC5 <<Staking>>
+        usecase "Earn Pablo LP\nTrading Fees" as UC6
+        usecase "Earn Pablo Protocol\nTrading Fees" as UC13 <<Staking>>
+        usecase "Set Pool Protocol\nFee Percentage" as UC7
+        usecase "Set PBLO Reward Rate" as UC9
+        usecase "Set PBLO Reward\nAllocation Percentage" as UC12
+    }
+    usecase "Stake PICA" as UC1 <<Staking>>
+    usecase "Earn PICA Reward" as UC3 <<Staking>>
+    usecase "Earn Other\nToken(s) Reward" as UC10 <<Staking>>
+    usecase "Set PICA Reward Rate" as UC8
+    usecase "Set PICA Reward\nAllocation Percentage" as UC11
+}
+
+p -up-> UC2
+p -up-> UC4
+p -up-> UC13
+s --> UC1
+s --> UC3
+s --> UC10
+lp --> UC4
+lp --> UC5
+lp --> UC6
+g -up-> UC7
+g -up-> UC8
+g -up-> UC9
+g -up-> UC11
+g -up-> UC12
+----
+
+== Requirements
+
+=== Pablo Liquidity Providers
+
+. LPs MUST be able to stake their LP tokens to earn rewards allocated for a particular pool.
+.. Rewards can be in terms of PBLO, PICA or any other tokens.
+.. Same pool can receive multiple types of tokens as rewards.
+. The system MUST support accumulating the LP share of Pablo trading fees.
+. Pablo trading fees(LP fee part) MUST be disbursed according to LP token share of each LP. Fees are accumulated towards increasing liquidity in a pool while allowing LPs to redeem the fee share with their LP tokens at a preferred time.
+
+=== PBLO Stakers
+
+. System MUST allow staking of PBLO. This must be implemented through the fNFT mechanism with multiple time period unlocks being possible for users.
+. The system MUST accumulate the rewards share for PBLO holders who stake PBLO token, out of the PBLO supply allocated for them.
+. The system MUST support accumulating the (stakers) reward part of the Pablo trading fees.
+. The system must support rewards being distributed on granular basis - e.g every 6 or 12 hours.
+. The users MUST be able to claim the rewards once distributed.
+. The system SHOULD support rewards in the form of fNFTs.
+
+=== PICA Stakers
+
+. System MUST allow staking of PICA. This must be implemented through the fNFT mechanism with multiple time period unlocks being possible for users.
+. The system MUST accumulate the rewards share for PICA holders who stake PICA token, out of the PICA supply allocated for them.
+. The system MUST support accumulating any token rewards other than PICA for PICA stakers.
+. The system must support rewards being distributed on granular basis - e.g every 6 or 12 hours.
+. The users MUST be able to claim the rewards once distributed.
```

Comment:
to claim distributed rewards eventually.

may be.

Reply:
The requirements are written as the action a party can take, the implementation details such as when they could be claimed are upto the method.

Reply:
so expectancy if there is reward then it is always possible to claim immediately? so automtic reward stops if there is no ability to claim?